src/cd-mapping-db.c

```diff
@@ -67,7 +67,7 @@ cd_mapping_db_open (CdMappingDb *mdb,
 		    GError  **error)
 {
 	CdMappingDbPrivate *priv = GET_PRIVATE (mdb);
-	g_autofree gchar *error_msg = NULL;
```

Comment:
Setting the value to `NULL` is initialization. And [`sqlite3_free`](https://www.sqlite.org/c3ref/free.html), `g_autofree` and [`g_free`](https://docs.gtk.org/glib/func.free.html) all handle `NULL` pointers just fine.

And actually, even if we did not initialize the value ourselves (although it is a good practice), [`sqlite3_exec`](https://www.sqlite.org/c3ref/exec.html) would always initialize it for us (as long as the function does not terminate before `sqlite3_exec` is called, which would be the case here).

I suspect the issue is that `sqlite3_malloc` uses some custom allocation code, which `g_free` (which uses system allocator) is unable to handle.